java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java

```diff
@@ -1236,6 +1237,17 @@ public void return_type_of_map_method() throws Exception {
     assertThat(result.symbol("test1").usages()).hasSize(1);
     assertThat(result.symbol("test2").usages()).hasSize(1);
     assertThat(result.symbol("test3").usages()).hasSize(1);
+  }
 
+  @Test
+  public void infer_method_invocation_return_type() throws Exception {
+    Result result = Result.createFor("CollectorsToList");
+    List<IdentifierTree> usages = result.symbol("foo").usages();
+    assertThat(usages).hasSize(1);
+    ExpressionTree arg = ((MethodInvocationTree) usages.get(0).parent()).arguments().get(0);
+    assertThat(arg.symbolType().is("java.util.List")).isTrue();
+    assertThat(((JavaType) arg.symbolType()).isParameterized());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Complete the assertion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2970)